common/lib/xmodule/xmodule/js/spec/capa/display_spec.coffee

```diff
@@ -802,3 +785,42 @@ describe 'Problem', ->
 
       # verify that codemirror textarea has correct `aria-describedby` attribute value
       expect($(CodeMirrorTextArea).attr('aria-describedby')).toEqual('cm-editor-exit-message-101 status_101')
+
+
+  describe 'show answer button', ->
+
+    radioButtonProblemHtml = readFixtures('radiobutton_problem.html')
+    checkboxProblemHtml = readFixtures('checkbox_problem.html')
+
+    beforeEach ->
+      @problem = new Problem($('.xblock-student_view'))
+
+      @clickShowAnswer = (ProblemHtml) =>
+        $('#input_example_1').replaceWith(ProblemHtml)
+        $('#input_1_1_1').attr('checked', true)
+        @problem.el.find('.show').trigger('click')
+        @problem.el.find('div.choicegroup label:nth-child(2)').addClass('choicegroup_correct')
+        @problem.el.find('div.choicegroup .choicegroup_correct').append('<span class="status correct"></span>')
+        @problem.el.find('.show').attr('disabled', 'disabled')
+
+      @submitCheckboxOrRadio = =>
+        # verify that by changing the selected answer, 'show answer button is no longer disabled'
+        expect(@problem.el.find('.show').attr('disabled')).not.toEqual('disabled')
+
+        # verify that displayed answer dispapears
+        expect(@problem.el.find('div.choicegroup')).not.toHaveClass('choicegroup_correct')
+
+        # verify that radio/checkbox label has no span having class '.status.correct'
+        expect(@problem.el.find('div.choicegroup')).not.toHaveAttr('span.status.correct')
+
+    it 'should become enabled after a radiobutton is selected', ->
```

Comment:
Please try to write flow of test here. How we managed to test this functionality?what each call is doing and why? just one line or two.